### PR TITLE
#138 Fallback Locale as array for cascading fallbacks

### PR DESCRIPTION
**Pull request by mmokross**
Implementation for different variants of locale fallbacks
- automatic implicit fallback on locales with territory and dialects
- without explicit fallback (false)
- on specific fallback locale (as already implemented already)
- array of fallback locales
- decision map with fallbacks as described in kazupon#138

### CODE REVIEW

**Comment by kazupon** [comment]
Thank you for your PR!
Great works! 👍 

I'm currently developing vue-i18n (a.k.a vue-i18n-next) for Vue 3.
https://github.com/intlify/vue-i18n-next
I'm going to release the alpha version soon.

This fallback locale feature is very nice!
I also need to give feedback to the vue-i18n-next.

I would like to release vue-i18n-next alpha version after that then merge your PR to the stable version vue-i18n@v8.x.

Please little wait! 🙇

**Comment by mmokross** [comment]
Cool!👍 No problem, take your time! 🙋‍♂️

File: gitbook/en/fallback.md [hunk 1]
```diff
@@ -33,3 +47,43 @@ The following will be the output:
...
+},
+```
+
+will result in the following fallback chains
+
+| locale | fallback chains |
+|--------|-----------------|
+| `'de-CH'`   | de-CH > fr > it > en > da |
+| `'de'`      | de > en > da |
+| `'zh-Hant'` | zh-Hant > zh-Hans > zh > en > da |
```
Comment (kazupon) [inline gitbook/en/fallback.md:84]: @mmokross 
In my understanding, I thin that not `zh-Hant > zh-Hans > zh > en > da`, but it should ``zh-Hant > zh-Hans > en > da`.
Right?

File: src/index.js [hunk 2]
```diff
@@ -232,6 +235,7 @@ export default class VueI18n {
 
   get fallbackLocale (): Locale { return this._vm.fallbackLocale }
   set fallbackLocale (locale: Locale): void {
+    this._vm.$set(this._vm, '_localeChainCache', new Map())
```
Comment (kazupon) [inline src/index.js:238]: I think that `_localeChainCache` don't need set Vue instance, because it's not used reactive by user.

Comment (kazupon) [reply src/index.js:238]:
```suggestion
    this._localeChainCache = new Map()
```

File: gitbook/en/fallback.md (hunk 1 above)
Comment (kazupon) [reply gitbook/en/fallback.md:84]: sorry, I misunderstood.

**Comment by kazupon** [comment]
Thank you very much!

**Pull request merged**